base-containers/base/inginious_container_api/utils.py

```diff
@@ -62,9 +62,12 @@ def start_ssh_server(ssh_user):
     execute_process(["/usr/sbin/sshd",
                     "-p", "22",
                     "-o", "PermitRootLogin={}".format(permit_root_login),
-                    "-o", "PasswordAuthentication=yes", "-o", "StrictModes=no", "-o",
-                    "AllowUsers={}".format(ssh_user)], internal_command=True, user=ssh_user)
+                    "-o", "PasswordAuthentication=yes", "-o", "StrictModes=no",
+                    "-o", "ForceCommand=echo LOGIN: Good luck !; script -q .ssh_logs; cp .ssh_logs student/.ssh_logs; echo LOGOUT: Good bye!",
```

Comment:
From the tests I did, this works well.
It allows the student to play within an interactive shell (launched via the "script" command) and closes the ssh connection when he leaves the shell. The log is written within the ssh session with worker permissions so the student is able to delete the .ssh_logs file (in the task directory) but should probably only code in the "student" directory so the chance of a sad rm * deleting it is low. Only when the student leaves the session, the file is copied into the student directory to be reflected within the grading_container.